dev/src/index.ts

```diff
@@ -349,11 +349,21 @@ export class Firestore {
   /**
    * Count of listeners that have been registered on the client.
    *
-   * The client can only be terminated when there are no registered listeners.
+   * The client can only be terminated when there are no pending writes or
+   * registered listeners.
    * @private
    */
   private registeredListenersCount = 0;
 
+  /**
+   * Number of pending operations on the client.
+   *
+   * The client can only be terminated when there are no pending writes or
+   * registered listeners.
+   * @private
+   */
+  private pendingOperationsCount = 0;
```

Comment:
Make this name bulk writer specific, since we are not sharing it with other logic. There could be many other pending operations (such as normal set()s).

Reply:
Renamed to `bulkWritersCount`